src/modules/dashboard/components/CreatePaymentDialog/CreatePaymentDialogForm.tsx

```diff
@@ -0,0 +1,394 @@
+  status,
+  values,
+}: Props & FormikProps<FormValues>) => {
+  const { formatMessage } = useIntl();
+
+  const { tokenAddress, amount } = values;
+  const fromDomain = values.fromDomain
+    ? parseInt(values.fromDomain, 10)
+    : ROOT_DOMAIN_ID;
+
+  const selectedToken = useMemo(
+    () => tokens.find((token) => token.address === values.tokenAddress),
+    [tokens, values.tokenAddress],
+  );
+
+  const tokenOptions = useMemo(
+    () =>
+      tokens.map(({ address, symbol }) => ({
+        value: address,
+        label: symbol || '???',
+      })),
+    [tokens],
+  );
+
+  const { walletAddress } = useLoggedInUser();
+
+  const fromDomainRoles = useTransformer(getUserRolesForDomain, [
+    colony,
+    walletAddress,
+    fromDomain,
+  ]);
+
+  const domainOptions = useMemo(
+    () =>
+      sortBy(
+        domains.map(({ name, ethDomainId }) => ({
+          value: ethDomainId.toString(),
+          label: name,
+        })),
+        ['value'],
+      ),
+
+    [domains],
+  );
+
+  const [
+    loadTokenBalances,
+    { data: tokenBalancesData },
+  ] = useTokenBalancesForDomainsLazyQuery();
+
+  useEffect(() => {
+    if (tokenAddress) {
+      loadTokenBalances({
+        variables: {
+          colonyAddress,
+          tokenAddresses: [tokenAddress],
+          domainIds: [fromDomain],
+        },
+      });
+    }
+  }, [colonyAddress, tokenAddress, fromDomain, loadTokenBalances]);
+
+  const fromDomainTokenBalance = useMemo(() => {
+    const token =
+      tokenBalancesData &&
+      tokenBalancesData.tokens.find(({ address }) => address === tokenAddress);
+    const from = getBalanceFromToken(token, fromDomain);
+    return from;
+  }, [fromDomain, tokenAddress, tokenBalancesData]);
+
+  useEffect(() => {
+    const errors: {
+      amount?: any;
+    } = {};
+
+    if (!selectedToken || !(amount && amount.length)) {
+      errors.amount = undefined; // silent error
+    } else {
+      const convertedAmount = bigNumberify(
+        moveDecimal(
+          amount,
+          getTokenDecimalsWithFallback(selectedToken.decimals),
+        ),
+      );
+      if (!convertedAmount.eq(0)) {
+        errors.amount = MSG.noAmount;
+      } else if (
+        fromDomainTokenBalance &&
+        fromDomainTokenBalance.lt(convertedAmount)
+      ) {
+        errors.amount = MSG.noBalance;
+      }
+    }
+
+    setErrors(errors);
+  }, [
+    amount,
+    fromDomain,
+    fromDomainRoles,
+    fromDomainTokenBalance,
+    selectedToken,
+    setErrors,
+  ]);
+
+  const userHasFundingPermission = userHasRole(
+    fromDomainRoles,
+    ColonyRole.Funding,
+  );
+  const userHasAdministrationPermission = userHasRole(
+    fromDomainRoles,
+    ColonyRole.Administration,
+  );
+  const userHasPermission =
+    userHasFundingPermission && userHasAdministrationPermission;
+  const requiredRoles: ColonyRole[] = [
+    ColonyRole.Funding,
+    ColonyRole.Administration,
+  ];
+
+  return (
+    <>
+      <FormStatus status={status} />
```

Comment:
What's the purpose of this for the current modal?

Reply:
Right, I think that this doesn't accomplish anything as of now so I'll remove it.